nucliadb/src/nucliadb/search/search/chat/ask.py

```diff
@@ -220,12 +223,19 @@ def _ndjson_encode(self, item: AskResponseItemType) -> str:
     async def _stream(self) -> AsyncGenerator[AskResponseItemType, None]:
         # First, stream out the predict answer
         first_chunk_yielded = False
+        first_reasoning_chunk_yielded = False
         with self.metrics.time("stream_predict_answer"):
             async for answer_chunk in self._stream_predict_answer_text():
-                yield AnswerAskResponseItem(text=answer_chunk)
-                if not first_chunk_yielded:
-                    self.metrics.record_first_chunk_yielded()
-                    first_chunk_yielded = True
+                if isinstance(answer_chunk, TextGenerativeResponse):
+                    yield AnswerAskResponseItem(text=answer_chunk.text)
+                    if not first_chunk_yielded:
+                        self.metrics.record_first_chunk_yielded()
+                        first_chunk_yielded = True
+                elif isinstance(answer_chunk, ReasoningGenerativeResponse):
+                    yield ReasoningAskResponseItem(text=answer_chunk.text)
+                    if not first_reasoning_chunk_yielded:
+                        self.metrics.record_first_reasoning_chunk_yielded()
+                        first_reasoning_chunk_yielded = True
```

Comment:
Can you add this? Is a little mypy hack to help us to maintain the code while adding new enum variants or types to conditionals
```suggestion
                if isinstance(answer_chunk, TextGenerativeResponse):
                    yield AnswerAskResponseItem(text=answer_chunk.text)
                    if not first_chunk_yielded:
                        self.metrics.record_first_chunk_yielded()
                        first_chunk_yielded = True
                elif isinstance(answer_chunk, ReasoningGenerativeResponse):
                    yield ReasoningAskResponseItem(text=answer_chunk.text)
                    if not first_reasoning_chunk_yielded:
                        self.metrics.record_first_reasoning_chunk_yielded()
                        first_reasoning_chunk_yielded = True
                else:
                    # This is a trick so mypy generates an error if this branch can be reached,
                    # that is, if we are missing some ifs
                    _a: int = "a"
```